WordPress/Classes/ViewRelated/Comments/CommentContentTableViewCell.swift

```diff
@@ -32,36 +35,128 @@ class CommentContentTableViewCell: UITableViewCell, NibReusable {
+    }()
+
+    /// Caches the HTML content, to be reused when the orientation changed.
+    private var htmlContentCache: String? = nil
+
     // MARK: Lifecycle
 
     override func awakeFromNib() {
         super.awakeFromNib()
         configureViews()
     }
 
+    override func prepareForReuse() {
+        onContentLoaded = nil
+        htmlContentCache = nil
+    }
+
     // MARK: Public Methods
 
-    func configure(with comment: Comment) {
+    /// Configures the cell with a `Comment` object.
+    ///
+    /// - Parameters:
+    ///   - comment: The `Comment` object to display.
+    ///   - onContentLoaded: Callback to be called once the content has been loaded. Provides the new content height as parameter.
+    func configure(with comment: Comment, onContentLoaded: ((CGFloat) -> Void)?) {
         nameLabel?.setText(comment.authorForDisplay())
         dateLabel?.setText(comment.dateForDisplay()?.toMediumString() ?? String())
 
-        if let authorURL = comment.authorURL() {
-            configureImage(with: authorURL)
+        if let avatarURL = URL(string: comment.authorAvatarURL) {
+            configureImage(with: avatarURL)
         } else {
             configureImageWithGravatarEmail(comment.gravatarEmailForDisplay())
         }
 
         updateLikeButton(liked: comment.isLiked, numberOfLikes: comment.numberOfLikes())
 
-        // TODO: Configure comment content
+        // configure comment content
+        guard let templateFormat = Self.htmlTemplateFormat else {
+            return
+        }
+
+        self.onContentLoaded = onContentLoaded
+
+        // remove empty HTML elements from the `content`, as the content often contains empty paragraph elements which adds unnecessary padding/margin.
+        // `rawContent` does not have this problem, but it's not used because `rawContent` gets rid of links (<a> tags) for mentions.
+        let sanitizedContent = comment.content
+            .trimmingCharacters(in: .whitespacesAndNewlines)
+            .replacingOccurrences(of: String.emptyElementRegexPattern, with: String(), options: [.regularExpression])
+        let htmlString = String(format: templateFormat, sanitizedContent)
+
+        self.htmlContentCache = htmlString
+        webView.loadHTMLString(htmlString, baseURL: Self.resourceURL)
 
         // TODO: Configure component visibility
     }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        // when screen orientation changed, reload the HTML content to ensure that the content height accounts for the new width.
+        guard let cachedContent = htmlContentCache,
+           traitCollection.verticalSizeClass != previousTraitCollection?.verticalSizeClass
+            || traitCollection.horizontalSizeClass != previousTraitCollection?.horizontalSizeClass else {
+            return
+        }
+
+        webViewHeightConstraint.constant = 1
+        webView.loadHTMLString(cachedContent, baseURL: nil)
```

Comment:
Should have passed `Self.resourceURL` here so the CSS file are properly loaded. Will fix this tomorrow!

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/17129/commits/550a4d2732c9643ee2cd23e60bb09dafdcba6101 as part of the refactor to `viewWillTransition` from `traitCollectionDidChange`.